src/radical/pilot/configs/resource_princeton.json

```diff
@@ -64,21 +64,21 @@
         "lrms"                        : "SLURM",
         "cores_per_node"              : 28,
         "gpus_per_node"               : 4,
-        "agent_scheduler"             : "NOOP",
+        "agent_scheduler"             : "CONTINUOUS",
         "agent_spawner"               : "POPEN",
         "agent_launch_method"         : "SSH",
         "task_launch_method"          : "SRUN",
         "mpi_launch_method"           : "SRUN",
-        "pre_bootstrap_0"             : ["module purge", 
+        "pre_bootstrap_0"             : ["module purge",
                                          "module load intel/18.0/64/18.0.3.222",
                                          "module load intel-mpi/intel/2018.3/64",
                                          "module load intel-python/2.7"
                                         ],
-        "default_remote_workdir"      : "$HOME",
+        "default_remote_workdir"      : "/scratch/gpfs/$USER/",
+        "virtenv"                     : "/scratch/gpfs/$USER/ve.rp",
+        "virtenv_mode"                : "use",
```

Comment:
Out of curiosity, what's different between 'create' and 'use'? Does 'create' create a new virtualenv if it doesn't exist whereas 'use' uses existing virtualenv?

Reply:
Good question :-)  `create` will create the ve if it does not exist - any subsequent uses of `create` will reuse the thus created ve.  `use` expects the ve to already exist, and will never attempt to create it.  We use this on resources where the automated ve creation fails.  In those cases, the user has to create the ve manually in advance.  On tiger, we don't have virtualenv installed, but also can't fetch virtualenv from the bootstrapper as the compute node have no direct internet link - thus the manual creation and `use`...